frontend/src/components/PrivateRoute.js

```diff
@@ -0,0 +1,7 @@
+import { Navigate, Outlet } from 'react-router-dom';
+
+function PrivateRoute() {
+    return !localStorage.getItem('token') ? <Navigate to='/signin' /> : <Outlet />;
```

Comment:
Hi, can you help us do a favor here thanks.

1. Create a new route in user-service that use the authenticateJwt function defined by lixue, can return a response telling you whether the token is valid. 

2. Then, in this file here you can send a GET request to user-service along with the cookie stored, using axios.

Refer to [this stackoverflow](https://stackoverflow.com/questions/43002444/make-axios-send-cookies-in-its-requests-automatically) to see how to do that.

3. The the res.success returned is true you navigate them to outlet, else to signin page